demes/demes.py

```diff
@@ -1012,12 +1029,8 @@ def _add_epoch(
             if end_size is None:
                 end_size = start_size
 
-        # XXX: use math.isclose()?
         if size_function is None:
-            if start_size == end_size:
```

Comment:
We still want it to default to "constant" if the start and end sizes are the same, right?

Reply:
That's definitely an option. I just changed the behaviour to match what Jerome suggested in #262. Defaulting to `exponential` is *slightly* simpler.

Reply:
I think I had misunderstood Jerome's suggestion before. I'm imagining software that only handle piecewise constant, so they check the size function of each epoch in a loaded graph, but if the default is to exponential, they'd fail even when start size == end size.

Reply:
I suppose. Did you have any specific software in mind here? Surely software which can only handle piecewise-constant models are able to partition each exponential epoch into `m` piecewise-constant subepochs?

Reply:
I'm thinking of PSMC/stairway plot type methods. I don't think we'd want to ask them to partition an epoch that is actually already constant but having a size function of "exponential". There are other methods that are more expensive to compute transitions for an exponential epoch as opposed to a constant size epoch, so they'd be interested in seeing the "constant" value if it actually is a constant size.

Reply:
The change here only affects the default value when reading a YAML file that has no `size_function` explicitly given. Any inference method that produces piecewise-constant epochs can specify `defaults: {epoch: {size_function: constant}}` in the YAML they output.

Reply:
Right - but in `moments` for example, if I read in a YAML file and want to compute the SFS, I create transition matrices for given sizes in each epoch. For constant sizes, these are computed once. But they need to be recomputed every time step of integration if the sizes are non-constant functions, and `moments` reads the `size_function` field to determine this.

Pretty sure users and other developers would also be confused to see epochs where the start and end times are equal, and then see that the size function defaulted to exponential when constant is an option.

Reply:
Ok, I'm happy to change this. Just wanted to talk it through in case I was missing something obvious. :)

Would you be satisfied with defaulting to "constant" `if not math.isclose(start_size, end_size)`, rather than `if start_size != end_size`? The spec wording could be something like: if size_function is not specified, implementations are expected to set the size_function to "constant" when start_size and end_size are the same (to within an implementation-defined numerical tolerance), and "exponential" otherwise.

Reply:
Ok, no problem! Yeah, I think that's reasonable. Do you mean something like `if math.isclose(start_size, end_size)` -> "constant", else "exponential"?

Reply:
> Ok, no problem! Yeah, I think that's reasonable. Do you mean something like `if math.isclose(start_size, end_size)` -> "constant", else "exponential"?

Ughh.. yes. Sorry!

Reply:
> I think I had misunderstood Jerome's suggestion before. I'm imagining software that only handle piecewise constant, so they check the size function of each epoch in a loaded graph, but if the default is to exponential, they'd fail even when start size == end size.

I understood that comment to mean "exponential if start != end".

Reply:
I think this is looking good!